src/Physics.hs

```diff
@@ -140,8 +95,65 @@ movePaddle game
         halfPaddle = paddleWidth / 2
         leftGameBorder = -(gameWidth / 2) + wallWidth / 2
         rightGameBorder = gameWidth / 2 - wallWidth / 2
+        paddleStep = 1
+
 
 
+--  | Update the ball velocity and bricks if the ball hits a brick
+bricksBounce :: Seconds   -- ^ Number of seconds since last update
+              -> Game     -- ^ current state pf the game
+              -> Game     -- ^ Game updated
+bricksBounce s game = case fst bc of
+      Nothing -> game
+      Just (vx, vy) -> game { bricks    = bricksUpdated
+                            , ballVel   = speedUp (vx / s, vy / s)
+                            , gameScore = addScore score
+                            }
+      where
+        bc = bricksCollision (vx * s, vy * s) (ballDots game) (bricks game)
+        (vx, vy) = ballVel game
+        bricksUpdated = snd bc
+        score = gameScore game
+
+-- | Detect collision on the paddle and change velocity and score
+paddleBounce :: Seconds -- ^ second since last update
+             -> Game    -- ^ current game state
+             -> Game    -- ^ game updated
+paddleBounce s game = case nws of
+        Nothing -> game
+        Just t -> game { ballVel = (nsX / s + (paddleV * 50) , nsY / s)}
+              where (nsX, nsY) = collisionToSpeed t (ballVX * s, ballVY * s)
+        where
+          paddleV = fst $ paddleVel $ paddle game
+          dots = ballDots game
+          (ballVX, ballVY) = ballVel game
+          nws = detectDotsCollision (ballVX * s, ballVY * s) dots $ paddleToRectangle $ paddle game
+
+-- | Detect collision on the walls and change ball velocity
+wallsBounce :: Seconds   -- ^ seconds since last update
+            -> Game     -- ^ current game state
+            -> Game     -- ^ game updated
+wallsBounce s game =
+      case collisions of
+          [] -> game
+          (x:xs) -> game { ballVel = collisionToSpeed x (ballVX, ballVY) }
+      where
+        dots = ballDots game
+        (ballVX, ballVY) = ballVel game
+        speed = (ballVX * s, ballVY * s)
+        gameWalls = [ (wallUpPos, gameWidth, wallWidth)
+                  , (wallLeftPos, wallWidth, gameHeight)
```

Comment:
Coma is misaligned with the above bracket.

Reply:
To avoid having to fix those things all the time, I tend to add a newline like this:

```haskell
gameWalls =
  [ (wallUpPos, gameWidth, wallWidth)
  , ...
  ]
```

This way, I can change the `gameWalls` identifier without having to re-indent the whole list.